$NetBSD: patch-ae,v 1.10 2021/12/29 23:03:21 dsainty Exp $

- ERROR: [check-portability.awk] => Found test ... == ...:

- automake-1.13 compat.

- allow a different default serial/parallel port to be specified
  which is useful for pkgsrc (since we have defaults for various
  platforms available and they're not the same for different
  MACHINE_ARCH's in NetBSD).

--- configure.ac.orig	2016-02-17 06:29:19.000000000 +0900
+++ configure.ac	2017-01-29 11:15:27.000000000 +0900
@@ -51,7 +51,7 @@ AC_PROG_AR
 AH_TEMPLATE([HAVE_YYLEX_DESTROY],
             [Define if lex/flex has yylex_destroy])
 # flex should have this
-if test "x$LEX" == xflex; then
+if test "x$LEX" = xflex; then
    AC_MSG_CHECKING([whether yylex_destroy is generated by flex])
    flex_version=`$LEX -V -v --version 2>/dev/null | $SED -e 's/^.* //'`
    case $flex_version in
@@ -365,32 +365,44 @@ AC_SUBST(DIST_SUBDIRS_AC, $DIST_SUBDIRS_
 # If a system doesn't have a PC style parallel, mark it as unknown.
 case $target in
 	i[[3456]]86-*-linux*|x86_64-*-linux*)
-		DEFAULT_PAR_PORT="/dev/parport0"
-		DEFAULT_SER_PORT="/dev/ttyS0"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"/dev/parport0"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/ttyS0"}
 		;;
 	*-*-linux*)
-		DEFAULT_PAR_PORT="unknown"
-		DEFAULT_SER_PORT="/dev/ttyS0"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"unknown"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/ttyS0"}
 		;;
 	i[[3456]]86-*-*freebsd*|amd64-*-*freebsd*)
-		DEFAULT_PAR_PORT="/dev/ppi0"
-		DEFAULT_SER_PORT="/dev/cuad0"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"/dev/ppi0"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/cuad0"}
 		;;
 	*-*-*freebsd*)
-		DEFAULT_PAR_PORT="unknown"
-		DEFAULT_SER_PORT="/dev/cuad0"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"unknown"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/cuad0"}
+		;;
+	*-*-dragonfly*)
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"/dev/ppi0"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/cuad0"}
+		;;
+	*-*-netbsd*)
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"unknown"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/tty00"}
 		;;
 	*-*-solaris*)
-		DEFAULT_PAR_PORT="/dev/printers/0"
-		DEFAULT_SER_PORT="/dev/term/a"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"/dev/printers/0"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/term/a"}
+		;;
+	*-*-msdos* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"lpt1"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"com1"}
+		;;
+	*)
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"unknown"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"unknown"}
 		;;
 	*-*-msdos* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
-		DEFAULT_PAR_PORT="lpt1"
-		DEFAULT_SER_PORT="com1"
 		;;
 	*)
-		DEFAULT_PAR_PORT="unknown"
-		DEFAULT_SER_PORT="unknown"
 		;;
 esac
 
